Tests/OpenCombineTests/PublisherTests/EncodeTests.swift

```diff
@@ -0,0 +1,77 @@
+//
+//  EncodeTests.swift
+//
+//
+//  Created by Joseph Spadafora on 6/21/19.
+//
+
+import XCTest
+
+#if OPENCOMBINE_COMPATIBILITY_TEST
+import Combine
+#else
+import OpenCombine
+#endif
+
+@available(macOS 10.15, *)
+final class EncodeTests: XCTestCase {
+    static let allTests = [
+        ("testEncodeWorks", testEncodeWorks),
+        ("testDemand", testDemand)
+    ]
+
+    private let jsonEncoder = JSONEncoder()
+    private let jsonDecoder = JSONDecoder()
+
+    func testEncodeWorks() throws {
+        // Given
+        let testValue = ["test": "TestDecodable"]
+        let subject = PassthroughSubject<[String: String], Error>()
+        let publisher = subject.encode(encoder: jsonEncoder)
+        let subscriber = TrackingSubscriberBase<Data, Error>()
+
+        // When
+        publisher.subscribe(subscriber)
+        subject.send(testValue)
+
+        // Then
+        guard case let .value(data) = subscriber.history[1] else {
+            XCTFail("No encoded data found")
+            return
+        }
```

Comment:
Let's compare the whole history here as well.

Reply:
I'm actually trying to get the `data` out of the value here to test for the decoding actually working.  I can add in a separate test for the history though.